src/remix/sdk-setup.ts

```diff
@@ -0,0 +1,509 @@
+import type { ExportNamedDeclaration, Program } from '@babel/types';
+
+// @ts-ignore - magicast is ESM and TS complains about that. It works though
+import type { ProxifiedModule } from 'magicast';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as url from 'url';
+import chalk from 'chalk';
+import { parse } from 'semver';
+
+// @ts-ignore - clack is ESM and TS complains about that. It works though
+import clack from '@clack/prompts';
+
+// @ts-ignore - magicast is ESM and TS complains about that. It works though
+import { builders, generateCode, loadFile, writeFile } from 'magicast';
+import * as recast from 'recast';
+import {
+  ERROR_BOUNDARY_TEMPLATE_V2,
+  HANDLE_ERROR_TEMPLATE_V2,
+} from './templates';
+import { PackageDotJson, getPackageVersion } from '../utils/package-json';
+
+export type PartialRemixConfig = {
+  unstable_dev?: boolean;
+  future?: {
+    v2_dev?: boolean;
+    v2_errorBoundary?: boolean;
+    v2_headers?: boolean;
+    v2_meta?: boolean;
+    v2_normalizeFormMethod?: boolean;
+    v2_routeConvention?: boolean;
+  };
+};
+
+const REMIX_CONFIG_FILE = 'remix.config.js';
+
+// Copied from sveltekit wizard
+function hasSentryContent(fileName: string, fileContent: string): boolean {
+  if (fileContent.includes('@sentry/remix')) {
+    clack.log.warn(
+      `File ${chalk.cyan(path.basename(fileName))} already contains Sentry code.
+Skipping adding Sentry functionality to ${chalk.cyan(
+        path.basename(fileName),
+      )}.`,
+    );
+
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Copied from sveltekit wizard
+ * We want to insert the init call on top of the file but after all import statements
+ */
+function getInitCallInsertionIndex(originalHooksModAST: Program): number {
+  // We need to deep-copy here because reverse mutates in place
+  const copiedBodyNodes = [...originalHooksModAST.body];
+  const lastImportDeclaration = copiedBodyNodes
+    .reverse()
+    .find((node) => node.type === 'ImportDeclaration');
+
+  const initCallInsertionIndex = lastImportDeclaration
+    ? originalHooksModAST.body.indexOf(lastImportDeclaration) + 1
+    : 0;
+  return initCallInsertionIndex;
```

Comment:
can we just run this in a for loop? More efficient than cloning, and iterating through twice.

```js
for (let x = originalHooksModAST.body.length - 1; x >= 0; x--) {
  if (originalHooksModAST.body[x].type === 'ImportDeclaration') {
    return x + 1;
  }
}

return 0;
```